### PR TITLE
add path hierarchy tokenizer docs

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add path hierarchy tokenizer docs

### Issues Resolved
Part of https://github.com/opensearch-project/documentation-website/issues/1483 addressed in this PR.

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas @varun-lodaya Please provide tech review approval to move this PR forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/tokenizers/path-hierarchy.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+---
+layout: default
+title: Path hierarchy
+parent: Tokenizers
+nav_order: 90
+---
+
+# Pattern tokenizer
```
Comment (natebower) [inline _analyzers/tokenizers/path-hierarchy.md:8]:
```suggestion
# Path hierarchy tokenizer
```

File: _analyzers/tokenizers/path-hierarchy.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+The `path_hierarchy` tokenizer is designed to tokenize file-system-like paths (or similar hierarchical structures) by breaking them down into tokens at each hierarchy level. This tokenizer is particularly useful when working with hierarchical data such as file paths, URLs, or any other delimited paths.
```
Comment (natebower) [inline _analyzers/tokenizers/path-hierarchy.md:10]:
```suggestion
The `path_hierarchy` tokenizer tokenizes file-system-like paths (or similar hierarchical structures) by breaking them down into tokens at each hierarchy level. This tokenizer is particularly useful when working with hierarchical data such as file paths, URLs, or any other delimited paths.
```

File: _analyzers/tokenizers/path-hierarchy.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+## Parameters
+
+The `path_hierarchy` tokenizer can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`delimiter` | Optional | String | Specifies the character used to separate path components. Default is `/`.
+`replacement` | Optional | String | Configures the character used to replace the delimiter in the tokens. Default is `/`.
+`buffer_size` | Optional | Integer | Specifies the buffer size. Default is `1024`.
+`reverse` | Optional | Boolean | If `true`, specifies to generate tokens in reverse order. Default is `false`.
```
Comment (natebower) [inline _analyzers/tokenizers/path-hierarchy.md:97]:
```suggestion
`reverse` | Optional | Boolean | If `true`, generates tokens in reverse order. Default is `false`.
```

File: _analyzers/tokenizers/path-hierarchy.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+`skip` | Optional | Integer | Specifies the number of initial tokens (levels) to skip when tokenizing. Default is `0`.
+
+## Example using delimiter and replacement parameters
+
+The following example configures custom `delimiter` and `replacement` parameters:
```
Comment (natebower) [inline _analyzers/tokenizers/path-hierarchy.md:102]:
```suggestion
The following example request configures custom `delimiter` and `replacement` parameters:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @AntonEliatra Editorial review. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**